code_generator/GdkCodeGenerator/Templates/UnityReaderWriterGenerator.tt

```diff
@@ -0,0 +1,71 @@
+<#@ template language="C#" #>
+<#@ output extension=".cs" #>
+<#
+    var generatedHeader = CommonGeneratorUtils.GetGeneratedHeader();
+    var componentDetails = GetComponentDetails();
+    var fieldDetailsList = GetFieldDetailsList();
+    var eventDetailsList = GetEventDetailsList();
+    var commandDetailsList = GetCommandDetailsList();
+#>
+<#= generatedHeader #>
+
+using Unity.Entities;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.MonoBehaviours;
+using Improbable.Worker;
+using Entity = Unity.Entities.Entity;
+
+namespace <#= qualifiedNamespace #>
+{
+    public partial class <#= componentDetails.ComponentName #>
+    {
+        [ComponentId(<#= unityComponentDefinition.Id #>)]
+        public class ReaderWriterCreator : IReaderWriterCreator
```

Comment:
Codegenerated ReaderWriterCreators allow us to create ReaderWriter impls without needing to use reflection. We only need to reflectively create instances of the creators on startup.